src/syft/core/common/event_loop.py

```diff
@@ -0,0 +1,115 @@
+# stdlib
+import asyncio
+import atexit
+import os
+from typing import Any as TypeAny
+from typing import Dict as TypeDict
+from typing import Optional
+
+# third party
+import nest_asyncio
+
+# syft relative
+from .environment import is_interactive
+from .environment import is_jupyter
+
+# Depending on where syft is imported we may need to get the existing event loop or
+# create a new one. In the case of Jupyter there is an existing event loop created
+# which we need to get otherwise our events might get blocked. If we import via CLI
+# there will be no event loop so we need to create one here and set it or the network
+# code will try to create its own and in some instances they wont match
+
+try:
+    # stdlib
+    from asyncio import get_running_loop  # noqa Python >=3.7
+except ImportError:  # pragma: no cover
+    # stdlib
+    from asyncio.events import _get_running_loop as get_running_loop  # pragma: no cover
+
+# first lets get any existing event loop, this will throw an exception in if there is
+# no event loop or will return none in older versions of python
+loop = None
+try:
+    loop = get_running_loop()
+except RuntimeError:
```

Comment:
Q: Print a warning? with the error?

Reply:
I think this only happens in python 3.7+ if there is no loop and in python 3.6 it returns None, so we just try to get it and if it fails we create a new one anyway.